doc/radosgw/compression.rst

```diff
@@ -32,7 +32,7 @@ recompressed.
 
 Compression settings apply to all new objects uploaded to buckets using this
 placement target. Compression can be disabled by setting the ``type`` to an
-empty string or ``none``.
+empty string or ``none``. The valid types are ``zlib`` and ``zstd``.
```

Comment:
there's a bullet list just above `Supported compression plugins include the following:`
https://github.com/ceph/ceph/blob/2996d132/doc/radosgw/compression.rst?plain=1#L13-L18

there's also a brotli plugin that i think is enabled by default

Reply:
Ah.  I hadn't thought to look above, assumed that inclusion on line 35 meant the choices weren't previously documented.  Does this mean that this change to line 35 is actually not needed?

Reply:
yeah, this part seems both redundant and inconsistent

Reply:
I suppose I missed that because we use two different terms; plugin and type. Should we uniformly use `type` since it's the term used to configure in the zonegroup?

We also have different compressors, the default original one, ISA-L, and QAT. Those are controlled with the params `compressor_zlib_isal` and `qat_compressor_enabled`.

It seems like we probably should have something like `compression_plugin` or `compressor` that can be set to `default`, `isal`, or `qat`?

We use the term "plugin" in other places, for EC (jerasure, isa, etc) and crypto via `plugin_crypto_accelerator` which can be `crypto_isal` (default) and `crypto_qat`.

Reply:
> I suppose I missed that because we use two different terms; plugin and type. Should we uniformly use type since it's the term used to configure in the zonegroup?

personally, i think the doc is pretty clear in its handling of plugin vs type

> The Ceph Object Gateway supports server-side compression of uploaded objects.
using any of the existing compression plugins.
...
Supported compression plugins include the following:
...
The compression ``type`` refers to the name of the compression plugin that will
be used when writing new object data.
...
Compression can be disabled by setting the ``type`` to an
empty string or ``none``.

but maybe it'd be simpler if we avoid mentioning plugin, and just list the supported types (including 'none')

Reply:
I think elsewhere in the docs we use "plugin", when describing compression and even EC. 

That said, I mainly wanted to be sure that if we enumerate them, we have a full list.  We've lacked that in other places in the past.

Reply:
> It seems like we probably should have something like compression_plugin or compressor that can be set to default, isal, or qat?

i don't think these belong with the zone's compression type, since that zone info is shared by all radosgws in the zone. hardware may differ between radosgw nodes, and ceph config seems like the best way to handle that

though i could see the orchestrators offering a choice between default/isal/qat and setting ceph configs accordingly